x/staging/posts/types/posts.go

```diff
@@ -130,16 +130,16 @@ func NewAttachment(uri, mimeType string, tags []string) Attachment {
 // Validate implements validator
 func (attachments Attachment) Validate() error {
 	if !commons.IsURIValid(attachments.URI) {
-		return fmt.Errorf("invalid uri provided")
+		return ErrInvalidAttachmentURI
```

Comment:
I think it's better to use a more generic `ErrInvalidPostAttachment` and specify the error here